.github/workflows/tests-e2e-npm.yml

```diff
@@ -0,0 +1,63 @@
+name: NPM Static checks
+on:
+  workflow_call:
+    inputs:
+      env_vars:
+        required: false
+        type: string
+        default: '{}'
+      build_tsoa:
+        required: false
+        type: boolean
+        default: false
+      db_migrate:
+        required: false
+        type: boolean
+        default: false
+      docker_compose_file:
+        required: false
+        type: string
+        default: docker-compose.yml
+permissions:
+  contents: write
+jobs:
+  e2e-tests:
+    name: Run e2e tests
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set environment variables from JSON
+        env:
+          ENV_VARS_JSON: "${{ inputs.env_vars }}"
+        run: |
+          echo "Parsing JSON environment variables..."
+          echo "$ENV_VARS_JSON" | jq -r 'to_entries|map("\(.key)=\(.value|tostring)")|.[]' >> $GITHUB_ENV
+      - name: Setup Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          buildkitd-flags: '--debug'
+      - name: Build e2e containers
+        uses: docker/bake-action@v5
+        with:
+          builder: ${{ steps.buildx.outputs.name }}
+          files: ${{ inputs.docker_compose_file }}
+      - name: Run e2e tests
+        run: docker compose -f ${{ inputs.docker_compose_file }} up --exit-code-from e2e-tests --abort-on-container-exit --quiet-pull
+        if: ${{ github.event.repository.name != 'veritable-ui' }}
+      - name: Run e2e tests - veritable UI
+        run: docker compose -f ${{ inputs.docker_compose_file }} up --exit-code-from e2e-tests --abort-on-container-exit --quiet-pull
+        env:
+          VERITABLE_COMPANY_PROFILE_API_KEY: ${{ secrets.COMPANIES_HOUSE_API }}
+          VERITABLE_E2E_OUT_DIR: ${{ runner.temp }}
+        if: ${{ github.event.repository.name == 'veritable-ui' }}
```

Comment:
This is pretty disgusting. I'm going t olook myself and see if I can come up with something better but this is fine for now I guess